java/src/com/ibm/streamsx/topology/generator/spl/OperatorGenerator.java

```diff
@@ -319,7 +319,19 @@ static void paramClause(JSONObject graphConfig, JSONObject op,
     static void parameterValue(JSONObject param, StringBuilder sb) {
         Object value = param.get("value");
         Object type = param.get("type");
-        if (value instanceof String && !PARAM_TYPES_TOSTRING.contains(type)) {
+        if ("submissionParameter".equals(type)) {
+            JSONObject jo = (JSONObject) value;
+            String splType = (String) jo.get("splType");
+            String name = SPLGenerator.stringLiteral((String) jo.get("name"));
```

Comment:
the value does need to be converted to a string literal so using the generator's method for that seemed logical/sensible.  though the value presumably won't/shouldn't ever have any chars that require the escapes that stringLiteral() currently handles, it doesn't do any harm.  i'll remove it if you think it's best w/o it.